src/util/mapbox.js

```diff
@@ -133,19 +133,122 @@ function formatUrl(obj: UrlObject): string {
     return `${obj.protocol}://${obj.authority}${obj.path}${params}`;
 }
 
-export class TurnstileEvent {
+class TelemetryEvent {
     eventData: { anonId: ?string, lastSuccess: ?number, accessToken: ?string};
-    queue: Array<number>;
-    pending: boolean
+    queue: Array<any>;
     pendingRequest: ?Cancelable;
 
     constructor() {
         this.eventData = { anonId: null, lastSuccess: null, accessToken: config.ACCESS_TOKEN};
         this.queue = [];
-        this.pending = false;
         this.pendingRequest = null;
     }
 
+    fetchEventData() {
+        const isLocalStorageAvailable = storageAvailable('localStorage');
+        const storageKey = `${telemEventKey}:${config.ACCESS_TOKEN || ''}`;
+
+        if (isLocalStorageAvailable) {
+            //Retrieve cached data
+            try {
+                const data = window.localStorage.getItem(storageKey);
+                if (data) {
+                    this.eventData = JSON.parse(data);
+                }
+            } catch (e) {
+                warnOnce('Unable to read from LocalStorage');
+            }
+        }
+    }
+
+    saveEventData() {
+        const isLocalStorageAvailable = storageAvailable('localStorage');
+        const storageKey = `${telemEventKey}:${config.ACCESS_TOKEN || ''}`;
+
+        if (isLocalStorageAvailable) {
+            try {
+                window.localStorage.setItem(storageKey, JSON.stringify(this.eventData));
```

Comment:
This would override any previously stored event data. In the case of a turnstile request finishing before the map load request, the `lastSuccess` property of the turnstile event would be removed, and more than one turnstile request may be processed. 

The event data could include separate objects for each event type or use a different key for each event type.